Source/JavaScriptCore/wasm/js/JSToWasm.cpp

```diff
@@ -214,6 +218,294 @@ void marshallJSResult(CCallHelpers& jit, const TypeDefinition& typeDefinition, c
+        jit.emitFunctionPrologue();
+
+        // saveJSEntrypointInterpreterRegisters
+        jit.subPtr(CCallHelpers::TrustedImmPtr(Wasm::JSEntrypointInterpreterCallee::SpillStackSpaceAligned), CCallHelpers::stackPointerRegister);
+
+#if CPU(ARM64) || CPU(ARM64E)
+        CCallHelpers::Address memBaseSlot { GPRInfo::callFrameRegister, -2 * static_cast<long>(sizeof(Register)) };
+        CCallHelpers::Address wasmInstanceSlot { GPRInfo::callFrameRegister, -3 * static_cast<long>(sizeof(Register)) };
+        jit.storePair64(GPRInfo::wasmBaseMemoryPointer, GPRInfo::wasmBoundsCheckingSizeRegister, memBaseSlot);
+        jit.store64(GPRInfo::wasmContextInstancePointer, wasmInstanceSlot);
+#elif CPU(X86_64)
+        CCallHelpers::Address boundsCheckSlot { GPRInfo::callFrameRegister, -1 * static_cast<long>(sizeof(Register)) };
+        CCallHelpers::Address memBaseSlot { GPRInfo::callFrameRegister, -2 * static_cast<long>(sizeof(Register)) };
+        CCallHelpers::Address wasmInstanceSlot { GPRInfo::callFrameRegister, -3 * static_cast<long>(sizeof(Register)) };
+        jit.store64(GPRInfo::wasmBoundsCheckingSizeRegister, boundsCheckSlot);
+        jit.store64(GPRInfo::wasmBaseMemoryPointer, memBaseSlot);
+        jit.store64(GPRInfo::wasmContextInstancePointer, wasmInstanceSlot);
+#else
+        CCallHelpers::Address wasmInstanceSlot { GPRInfo::callFrameRegister, -1 * static_cast<long>(sizeof(Register)) };
+        jit.store32(GPRInfo::wasmContextInstancePointer, wasmInstanceSlot);
+#endif
+
+        {
+            // Callee[cfr]
+            CCallHelpers::Address functionSlot { GPRInfo::callFrameRegister, sizeof(CPURegister) * 2 + sizeof(Register) };
+            jit.loadPtr(functionSlot, GPRInfo::regWS0);
+        }
+
+        {
+            CCallHelpers::Address calleeSlot { GPRInfo::regWS0, WebAssemblyFunction::offsetOfInterpreterCallee() };
+            jit.loadPtr(calleeSlot, GPRInfo::regWS0);
+        }
+
+#if ASSERT_ENABLED
+        {
+            CCallHelpers::Address identSlot { GPRInfo::regWS0, JSEntrypointInterpreterCallee::offsetOfIdent() };
+            jit.load32(identSlot, GPRInfo::regWS1);
+            jit.move(MacroAssembler::TrustedImm32(0xBF), GPRInfo::regWA0);
+            auto ok = jit.branch32(CCallHelpers::Equal, GPRInfo::regWS1, GPRInfo::regWA0);
+            jit.breakpoint();
+            ok.link(&jit);
+        }
+#endif
+
+        // Allocate stack space (no stack check)
+        {
+            CCallHelpers::Address frameSlot { GPRInfo::regWS0, JSEntrypointInterpreterCallee::offsetOfFrameSize() };
+            jit.load32(frameSlot, GPRInfo::regWS1);
+            jit.subPtr(GPRInfo::regWS1, CCallHelpers::stackPointerRegister);
+        }
+
+        // Prepare frame
+        jit.setupArguments<decltype(operationJSToWasmEntryWrapperBuildFrame)>(CCallHelpers::stackPointerRegister, GPRInfo::callFrameRegister);
+        jit.callOperation<OperationPtrTag>(operationJSToWasmEntryWrapperBuildFrame);
```

Comment:
This can throw an error. So JS exception handling is necessary.

Reply:
I don't think it does throw actually.